packages/btcindexer/src/btcblock.ts

```diff
@@ -1,49 +1,57 @@
 import { Block } from 'bitcoinjs-lib';
 
+/**
+ * Extends the Block type to include raw Buffer and height
+ */
 export interface ExtendedBlock extends Block {
+	height: number;
 	raw: Buffer;
 }
 
-async function streamToBuffer(body: ReadableStream | null): Promise<Buffer> {
+/**
+ * Defines the structure of the JSON object we expect from the relayer
+ */
+interface BlockPayload {
+	height: number;
+	rawBlockHex: string;
+}
+
+async function streamToJson(body: ReadableStream | null): Promise<BlockPayload[]> {
 	if (!body) {
-		return Buffer.alloc(0);
+		return [];
 	}
 
 	const reader = body.getReader();
-	const chunks: Uint8Array[] = [];
+	const decoder = new TextDecoder();
+	let jsonString = ``;
 	while (true) {
 		const { done, value } = await reader.read();
 		if (done) break;
-		chunks.push(value);
+		jsonString += decoder.decode(value, { stream: true });
 	}
 
-	return Buffer.concat(chunks);
+	return JSON.parse(jsonString);
 }
 
-function parseBlocksFromBuffer(buffer: Buffer): ExtendedBlock[] {
```

Comment:
Can we do parsing directly from the body stream, without converting to JSON array? That sounds inefficient, especially when the blocks can be 4MB

Reply:
created an issue for it